crates/core/src/models/mod.rs

```diff
@@ -234,3 +239,60 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
         Ok(block.into())
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(&self, _client: &dyn KakarotClient) -> Result<Log, KakarotClientError> {
+        // let starknet_block_id = StarknetBlockId::Tag(BlockTag::Latest);
+        // let address = client.get_evm_address(&self.0.from_address, &starknet_block_id).await.unwrap();
```

Comment:
you can create a getter for from_address instead of self.0.from_address if you want

Reply:
i figured since it was only two getters and i dont need to dispatch, it would be fine to manually just do the self.0.<field>

wdyt? sound reasoning?

Reply:
sounds fair